clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1155,14 +1162,21 @@ public ClientRequest newClientRequest(String nodeId,
         return newClientRequest(nodeId, requestBuilder, createdTimeMs, expectResponse, defaultRequestTimeoutMs, null);
     }
 
+    // visible for testing
+    int nextCorrelationId() {
+        if (SaslClientAuthenticator.isReserved(correlation))
+            correlation = 0;
```

Comment:
I think we were using negative values as well before. Perhaps we should use `correlation = SaslClientAuthenticator.MAX_RESERVED_CORRELATION_ID + 1` to avoid assuming here that zero is not reserved?

Reply:
you are totally right! I will roger that!